src/azure-cli/azure/cli/__main__.py

```diff
@@ -96,10 +96,12 @@ def cli_main(cli, args):
                         else:
                             upgrade_exit_code = subprocess.call(cmd, shell=platform.system() == 'Windows')
                     else:
-                        import os
-                        devnull = open(os.devnull, 'w')
                         cmd.append('-y')
-                        upgrade_exit_code = subprocess.call(cmd, shell=platform.system() == 'Windows', stdout=devnull)
+                        upgrade_exit_code = subprocess.Popen(
```

Comment:
`subprocess.call` and `subprocess.Popen` are not the same. `subprocess.Popen` is not waited and doesn't return the exit code. See https://docs.python.org/3/library/subprocess.html

I would suggest not changing it.